tests/unit/test_project.py

```diff
@@ -66,6 +66,22 @@ def test_project_by_external_id_succeeds(self):
         actual_project = self.tamr.projects.by_external_id(self.project_external_id)
         self.assertEqual(self.project_json[0], actual_project._data)
 
+    @responses.activate
+    def test_project_by_name__raises_when_not_found(self):
+        responses.add(responses.GET, self.project_list_url, json=[])
+        auth = UsernamePasswordAuth("username", "password")
+        tamr = Client(auth)
+        with self.assertRaises(KeyError):
+            tamr.projects.by_name(self.project_name)
+
+    @responses.activate
+    def test_dataset_by_name_succeeds(self):
```

Comment:
No need for `_succeeds` suffix, since success is assumed for all tests without a `_raises` suffix

```suggestion
    def test_dataset_by_name(self):
```